framework/src/Volo.Abp.AspNetCore.Mvc.UI.Bootstrap/TagHelpers/CacheScripts/AbpApplicationConfigurationScriptTagHelperService.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Html;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+using Volo.Abp.AspNetCore.Mvc.ApplicationConfigurations;
+using Volo.Abp.AspNetCore.Mvc.ControllerScriptCacheItem;
+using Volo.Abp.Caching;
+using Volo.Abp.Json;
+
+namespace Volo.Abp.AspNetCore.Mvc.UI.Bootstrap.TagHelpers.CacheScripts;
+
+public class AbpApplicationConfigurationScriptTagHelperService : AbpTagHelperService<AbpApplicationConfigurationScriptTagHelper>
+{
+    protected IAbpApplicationConfigurationAppService AbpApplicationConfigurationAppService { get; }
+    protected IJsonSerializer JsonSerializer { get; }
+    protected IDistributedCache<AbpControllerScriptCacheItem> Cache { get; }
```

Comment:
Do not use `IDistributedCache`. Distributed cache is not proper to this use case;

* It is an heavy operation for such a large data.
* We are calculating it in every request, so we don't need to share it with other application instances.
* It may conflict with different versions of the same application since they will use the same cache item (actually, here all applications are using the same cache item, which is worse).

This comment is valid for other scripts too...

We can use IMemoryCache. However, if we use a web farm, the request will probably go to another server, so will be no meaning to cache it. In this case, we may consider to not cache it and always re-calculate. In this case, it will make a double calculation. I couldn't find an ideal solution yet. We should re-think that.